ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -619,4 +624,26 @@ static void maybeCreateProcessingLogStream(
 
     commandQueue.enqueueCommand(configured.get());
   }
+
+  private static void maybeConfigureRocksDBConfigSetter(final KsqlConfig ksqlConfig) {
+    final Map<String, Object> streamsProps = ksqlConfig.getKsqlStreamConfigProps();
+    if (streamsProps.containsKey(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG)) {
+      final Class<?> clazz =
+          (Class) streamsProps.get(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG);
+
+      final Method method;
+      try {
+        method = clazz.getMethod("configure", Map.class);
+      } catch (NoSuchMethodException e) {
+        // Nothing to configure
+        return;
```

Comment:
If the config is there, but the class does not have the correct method, then we should throw an appropriate exception, not silently swallow it.

Reply:
It's valid for the user to pass a different `RocksDBConfigSetter` that does not implement the static `configure()` method, and we want that to succeed. See the unit test `shouldSkipConfiguringRocksDBConfigSetterWithoutConfigureMethod()` below.

Reply:
The problem with this is that the UX is bad if the user _wants_ to have a configure method, but has the signature wrong.  It just silently fails.

I think a better UX is to require it, even if its a no-op, and then hard fail if it does not exist.

Reply:
> I think a better UX is to require it, even if its a no-op, and then hard fail if it does not exist.

This would break backwards compatibility if a user is already running with a `RocksDBConfigSetter` that doesn't implement the `configure()` method. If such a user upgraded their KSQL server, the server would fail to start unless they updated their `RocksDBConfigSetter` to include a no-op `configure()` method. Doesn't seem like great UX either, but I see your point.

Regardless, I believe this discussion is moot in light of your suggestion to perform the static configuration via `streamsConfig.getConfiguredInstance(...)` rather than checking for a static `configure()` method. I'll go with that approach instead.

Reply:
Discussed with @big-andy-coates offline: turns out using `getConfiguredInstance(...)` isn't a viable approach since `ksqlConfig.getConfiguredInstance(...)` fails the check within `getConfiguredInstance()` that the key is present in the `ConfigDef`, and we can't easily create a `StreamsConfig` from the `KsqlConfig` as its missing required configs such as `application.id`. In light of this, we'll opt to check whether the `RocksDBConfigSetter` implements `Configurable` or not (and call `configure()` if so), rather than call `getConfiguredInstance()`.